### ISSUE TITLE
Softmax in the last layer

### ISSUE THREAD

**Issue by cijerezg** [huyvnphan/PyTorch_CIFAR10#12]
Did you use a Softmax in the very last layer? I see that last layer is a fully connected (nn.Linear).

**Comment by cijerezg** [comment]
I didn't realize I just needed to apply that extra separately.

**Comment by kfarivar** [comment]
I have this problem too. I stil don't see any softmax used on the output, so I get this error when testing the pretrained network.
`python train.py --test_phase 1 --pretrained 1 --classifier resnet18`

`The `preds` should be probabilities, but values were detected outside of [0,1] range.`
which is a reasonable error. :)

this the part of code that is problematic:
[https://github.com/huyvnphan/PyTorch_CIFAR10/blob/7a6e1f90840bb201b7f6006adc0a1e3e7b6b5ccb/module.py#L44](url)
